spec/registry.md

```diff
@@ -619,6 +619,23 @@ All other values produce an _Invalid Expression_ error.
 > or the type `com.ibm.icu.util.CurrencyAmount` can be used to set the currency and related
 > options (such as the number of fraction digits).
 
+### Non-Negative Integer Options
+
+Some _options_ of number _functions_ are defined to take a "non-negative integer".
+Implementations of number _functions_ use these _options_ to control aspects of numeric display
+such as the number of fraction, integer, or significant digits.
+
+A "non-negative integer" is an _option_ value that the _function_ evaluates as an integer 
+greater than or equal to zero.
```

Comment:
```suggestion
A "non-negative integer" is an _option_ value that can be coerced to an integer
that is greater than equal to zero,
according to implementation-specific coercion rules.
```

Maybe this helps address my other comment too: the subsequent text could say that even though the coercion rules are implementation-specific, they MUST at least be defined so that a stringified integer can be coerced to an integer.

Reply:
I like this text. I originally thought about using the word "coerce" here. But that gets us back into the type game. I also added the bits about "... that the _function_ evaluates..." to make clear that it is **_not_** message format that is doing the evaluation/coercion (or at least not-necessarily the MF code).

We could say "implementation-specific coercion rules", although I think we want implementations to accept a sequence of ASCII digits (so that _message_ are portable). It seems dubious that anyone will get this wrong, given that one can have  a a number between `0` and maybe around `32` inclusive that produces some kind of meaningful result. (I mean, yeah, technically you can have "max int" digits, but that is a stupendously sized number and won't fit on most screens....... or print on less than a ream of paper 😁). With that in mind, I basically want us to say: "some implementation defined number type with a non-negative value or one of these ASCII digit strings". All else is rubbish.

Reply:
> I like this text. I originally thought about using the word "coerce" here. But that gets us back into the type game. I also added the bits about "... that the _function_ evaluates..." to make clear that it is **_not_** message format that is doing the evaluation/coercion (or at least not-necessarily the MF code).
> 

Right, I think I understand what you were trying to do, but "evaluates" is a bit too vague for my tastes. I understand your objection to "coerce", though.

> We could say "implementation-specific coercion rules", although I think we want implementations to accept a sequence of ASCII digits (so that _message_ are portable). It seems dubious that anyone will get this wrong, given that one can have a a number between `0` and maybe around `32` inclusive that produces some kind of meaningful result. (I mean, yeah, technically you can have "max int" digits, but that is a stupendously sized number and won't fit on most screens....... or print on less than a ream of paper 😁). With that in mind, I basically want us to say: "some implementation defined number type with a non-negative value or one of these ASCII digit strings". All else is rubbish.

Well, if we really want these to be finite enumerations, that's a lot easier :) Is there anything wrong with saying that the input MUST be either a string in `{"1", "2", ..., "32"}` or an integer in the interval [1, 32]?

Reply:
Different implementations will have different max values for these options. I didn't want to put a hard definition in. We could have a note that says "expect a small number". I would be surprised, for this particular set of options, if any implementation supported a value with more than two digits, but I don't think we need to prevent it here per-se.

There is a buffer-overflow attack from our syntax. This is valid:

```
{$n :number maximumFractionDigits=123456789123456789123456789123456789} <- plus more digits if you want
```